### ISSUE TITLE
Ignore SSL certificate errors

### ISSUE THREAD

**Issue by XP1** [xroche/httrack#150]
There needs to be an option to ignore SSL certificate errors, like wget's `--no-check-certificate`.

**Comment by knowmad** [comment]
Agreed. This may address the problem I just reported to the Forums at https://forum.httrack.com/readmsg/36334/index.html

**Comment by laurent-simon** [comment]
It is not possible to copy a web site when it's SSL certificate is expired. Sometimes, it is in a such situation that you need to take a copy of a site because it is not maintained anymore. The certificates are no renewed there is some risk that the site will disappear later, but you can't copy it!

In such a situation, where it's usage is imperative, httrack become totally useless. An option should be available to allow to ignore certificates errors.

**Comment by leonstafford** [comment]
[This forum link](http://forum.httrack.com/readmsg/38444/38178/index.html) suggests you could set the clock back to when the certificate was valid. May work for a single site, but if needing to mirror external https sites in same process, those certs may not have been valid at that date.

**Comment by johnwait** [comment]
Came here looking for a solution to archiving certificate-expired websites, but quickly became disillusioned seeing no progress had been made since last comment.

However, after testing and some successful runs, I can suggest a new option (more workaround than solution), for scenarios at least where HTTrack can be used on Windows platforms: the ["RunAsDate" utility](https://www.nirsoft.net/utils/run_as_date.html) by NirSoft.

Using the latest available version (1.41) on a Windows 10 x64 computer, I successfully archived a (real, single-domain) website with a certificate expired since June 2025, that I needed preserved, by starting WinHTTrack through the aforementioned RunAsDate by using a date & time just prior to the cert's expiration (in my case, the early hours of June 1st, 2025).

No messing with the system date, no proxy required or involved, and as far as I can tell, no SSL/TLS errors encountered, or at the very least, reported.

A scenario involving multiple domains would certainly make it harder, maybe impossible, to pick a suitable date & time that is covered by all currently-issued certificates; in such cases, proxying techniques would probably be required.

<img width="531" height="588" alt="Image" src="https://github.com/user-attachments/assets/bb58eaa9-dbf0-419d-815a-36275fe7d92d" />

**Comment by xroche** [comment]
Re-tested against current master and this no longer reproduces. I built from master (linked against OpenSSL 3.5.6) and mirrored SSL.com's expired and revoked sample sites. https://expired-rsa-dv.ssl.com/ and https://revoked-rsa-dv.ssl.com/ both copied fine, fetching the real pages with no errors.

The reason is that the OpenSSL client path never verifies the server certificate: there's no `SSL_CTX_set_verify` and no `SSL_get_verify_result` check anywhere in the source. Expired, revoked, self-signed, and CN-mismatch certs are all accepted by default, so what's requested here is already how it behaves and there's nothing to disable.

Closing as resolved for the OpenSSL builds (Linux, macOS, and the OpenSSL-based Windows build).

One caveat: older Windows builds on a different TLS backend can still reject expired certs, which is what the clock-rollback and RunAsDate workarounds in this thread were for. Moving those to the OpenSSL path is a separate follow-up, tracked on the Windows side.
